### PR TITLE
feat(cli): add support for custom trace categories

### LINKED ISSUE #1663: WPT compat: include additional trace categories
LH is missing the following trace categories captured by WPT:
```
+      "v8.execute",
+      "devtools.timeline.frame",
+      "devtools.timeline.stack",
+      "disabled-by-default-blink.feature_usage",
```

### PR DESCRIPTION
**Pull request by patrickhulce**
fixes #1663

### CODE REVIEW

File: lighthouse-core/gather/driver.js [hunk 1]
```diff
@@ -570,9 +570,14 @@ class Driver {
       });
   }
 
-  beginTrace() {
+  /**
+   * @param {{traceCategories: string=}=} flags
+   */
+  beginTrace(flags) {
+    const additionalCategories = flags && flags.traceCategories && flags.traceCategories.split(',');
+    const traceCategories = new Set(this._traceCategories.concat(additionalCategories || []));
```
Comment (paulirish) [inline lighthouse-core/gather/driver.js]: i kinda prefer moving this ` || []` to the above statement, so we can rely on it as an array.

Comment (paulirish) [inline lighthouse-core/gather/driver.js]: let's just drop in `const _uniq = arr => Array.from(new Set(arr));` and use it so we're not tossing this type from Set back to array a few lines later.

File: lighthouse-cli/bin.ts [hunk 2]
```diff
@@ -78,6 +79,7 @@ const cliFlags = yargs
     'save-artifacts': 'Save all gathered artifacts to disk',
     'list-all-audits': 'Prints a list of all available audits and exits',
     'list-trace-categories': 'Prints a list of all required trace categories and exits',
+    'trace-categories': 'Additional categories to capture with the trace (comma-delimited).',
```
Comment (paulirish) [inline lighthouse-cli/bin.ts]: how about `addl-trace-categories` ?  to clarify that this doesn't overwrite the defaults

File: lighthouse-core/gather/driver.js (hunk 1 above)
Comment (patrickhulce) [inline lighthouse-core/gather/driver.js]: done

File: lighthouse-cli/bin.ts (hunk 2 above)
Comment (patrickhulce) [inline lighthouse-cli/bin.ts]: `additional-` but done :)

File: lighthouse-core/gather/driver.js [hunk 3]
```diff
@@ -570,9 +572,15 @@ class Driver {
       });
   }
 
-  beginTrace() {
+  /**
+   * @param {{additionalTraceCategories: string=}=} flags
+   */
+  beginTrace(flags) {
```
Comment (ebidel) [inline lighthouse-core/gather/driver.js]: nit: could use some fancy default params: `flags = {additionalTraceCategories: ''}` 
then: `const additionalCategories = flags.additionalTraceCategories.split(',').filter(String);`

Comment (patrickhulce) [inline lighthouse-core/gather/driver.js]: true, but would still need the additionalTraceCategories check so just gets rid of `flags &&` in favor of ` = {additionalTraceCategories: ''}` gotta save those bytes you know ;)

**Pull request merged**